tests/serialize/runstate/dynamodb_state_store_test.py

```diff
@@ -263,11 +263,36 @@ def test_delete_item(self, store, small_object, large_object):
         pairs = list(zip(keys, (value for i in range(len(keys)))))
         store.save(pairs)
 
+        store._consume_save_queue()
         for key, value in pairs:
             store._delete_item(key)
-
         for key, value in pairs:
-            assert_equal(store._get_num_of_partitions(key), 0)
+            num_partitions, num_json_val_partitions = store._get_num_of_partitions(key)
+            assert_equal(num_partitions, 0)
+            assert_equal(num_json_val_partitions, 0)
+
+    def test_delete_item_with_json_partitions(self, store, small_object, large_object):
+        key = store.build_key("job_state", "test_job")
+        value = small_object
+
+        store.save([(key, value)])
+        store._consume_save_queue()
+
+        num_partitions, num_json_val_partitions = store._get_num_of_partitions(key)
+        assert num_partitions > 0
+        assert num_json_val_partitions > 0
```

Comment:
just curious: how come we're not asserting an exact # of partitions?

Reply:
I think it potentially more finicky/brittle and ultimately all I care about here is that we have partitions. Partition amounts might change as we update the max amount back to 400kb post-pickle + I've fiddled with the size of large_object a few times while working on other parts of Tron.

Reply:
gotcha - makes sense :)